server/src/main/java/io/crate/execution/ddl/tables/CreateTableRequest.java

```diff
@@ -83,21 +155,91 @@ public TimeValue ackTimeout() {
 
     public CreateTableRequest(StreamInput in) throws IOException {
         super(in);
-        if (in.readBoolean()) {
-            createIndexRequest = new CreateIndexRequest(in);
+        if (in.getVersion().onOrAfter(Version.V_5_4_0)) {
+
+            AddColumnRequest request = readColsToAdd(in);
+            this.relationName = request.relationName();
+            this.colsToAdd = request.references();
+            this.checkConstraints = request.checkConstraints();
+            this.pKeyIndices = request.pKeyIndices();
```

Comment:
`AddColumnRequest` as a temporal holder of value here - maybe just duplicate streaming code? Can at least reuse `writeColsToAdd` as we don't use intermediate objects there.